govc/vm/guest/guest.go

```diff
@@ -71,43 +69,17 @@ func (flag *GuestFlag) Process(ctx context.Context) error {
 }
 
 func (flag *GuestFlag) Toolbox() (*toolbox.Client, error) {
-	pm, err := flag.ProcessManager()
-	if err != nil {
-		return nil, err
-	}
-
-	fm, err := flag.FileManager()
-	if err != nil {
-		return nil, err
-	}
-
 	vm, err := flag.VirtualMachine()
 	if err != nil {
 		return nil, err
 	}
 
-	family := ""
-	var props mo.VirtualMachine
-	err = vm.Properties(context.Background(), vm.Reference(), []string{"guest.guestFamily", "guest.toolsInstallType"}, &props)
+	c, err := flag.Client()
 	if err != nil {
 		return nil, err
 	}
-	if props.Guest != nil {
-		family = props.Guest.GuestFamily
-		if family == string(types.VirtualMachineGuestOsFamilyOtherGuestFamily) {
-			if props.Guest.ToolsInstallType == string(types.VirtualMachineToolsInstallTypeGuestToolsTypeMSI) {
-				// The case of Windows version not supported by the ESX version
-				family = string(types.VirtualMachineGuestOsFamilyWindowsGuest)
-			}
-		}
-	}
 
-	return &toolbox.Client{
-		ProcessManager: pm,
-		FileManager:    fm,
-		Authentication: flag.Auth(),
-		GuestFamily:    types.VirtualMachineGuestOsFamily(family),
-	}, nil
+	return toolbox.NewClient(context.TODO(), c, vm, flag.Auth())
```

Comment:
nit: in other functions used in this file you declare `ctx` as variable. Might want to do the same for consistency.